include/cuco/detail/open_addressing/open_addressing_impl.cuh

```diff
@@ -814,6 +827,43 @@ class open_addressing_impl {
   [[nodiscard]] constexpr storage_ref_type storage_ref() const noexcept { return storage_.ref(); }
 
  private:
+  /**
+   * @brief Counts the occurrences of keys in `[first, last)` contained in the container
+   *
+   * @note If `IsOuter` is `true`, the occurrence of a non-match key is 1. Else, it's 0.
+   *
+   * @tparam IsOuter Flag indicating whether it's an outer count or not
+   * @tparam Input Device accessible input iterator
+   * @tparam Ref Type of non-owning device container ref allowing access to storage
+   *
+   * @param first Beginning of the sequence of keys to count
+   * @param last End of the sequence of keys to count
+   * @param stream CUDA stream used for count
+   *
+   * @return The sum of total occurrences of all keys in `[first, last)`
+   */
+  template <bool IsOuter, typename InputIt, typename Ref>
+  [[nodiscard]] size_type count(InputIt first,
+                                InputIt last,
+                                Ref container_ref,
+                                cuda_stream_ref stream) const noexcept
+  {
+    auto const num_keys = cuco::detail::distance(first, last);
+    if (num_keys == 0) { return 0; }
+
+    auto counter =
+      detail::counter_storage<size_type, thread_scope, allocator_type>{this->allocator()};
+    counter.reset(stream);
+
+    auto const grid_size = cuco::detail::grid_size(num_keys, cg_size);
```

Comment:
Since `count` and `count_outer` are basically reductions, using a grid stride loop might help with performance since it reduces the number of global updates. Not an action item for this PR, but I think it would be helpful to introduce a facility that calculates the optimal grid size based on the given SM count.

Reply:
`cuda::launch` is of course preferred but given that it might take another major version or so to be feature complete we might want to fall back to using the ol' driver API. I would say the only kernels that would benefit from a smaller grid in cuco are those that flush staging buffers from shmem to gmem or do some sort of reduction. So this would be a good place to start the investigation.